app/Dockerfile

```diff
@@ -1,11 +1,10 @@
 FROM python:3.6
 
-RUN pip install -U pip && pip install \
-    Flask==0.12.2 \
-    redis==2.10.6 \
-    uWSGI==2.0.15
-
 WORKDIR /code
+COPY . /code
```

Comment:
It's better copying the application code after the installation of the dependencies. In this way the Python package chache won't be invalidated if you change a single line in the app.

Reply:
But I agree to move the dependency in a proper file.
```
COPY requirements.txt /code
RUN pip install...
COPY . /code
```